format/index.md

```diff
@@ -785,6 +794,15 @@ include additional [fields] in the response.
 GET /articles?include=author&fields[articles]=title,body&fields[people]=name
 ```
 
+If both query parameters `include` and `fields[TYPE]` are provided by the client, 
+values of `TYPE` **MUST** either refer to the primary data or be listed in 
+`include`, otherwise the server **MUST** return `400 Bad Request`.
```

Comment:
This PR is only discussing the situation when the client provides both `fields[TYPE]` and `include`. Sparse fieldsets operate "on a per-type basis", not on per-relationship-name basis (the latter would be impossible: if we allow to specify different sparse filedsets for the same type, referred to via different relationships, it may result in a situation when we have two different fildsets requirements for a resource with the same `(type,id)`). If the client does "not know the members' types in the response", how could it require specific `fields[TYPE]` for the unknown `TYPE`? 

I believe there is no intrinsic problem here. The server knows which types constitute a particular heterogeneous relationships, so it is capable of performing the verification. The client apparently knows that too, since it specified `fields[TYPE]` in the first place. IMHO, this clause is not more and not less mandatory than other clauses which require the server to return specific errors if the input it receives makes no sense. Please correct me if I am wrong.

I have corrected the wording to account for the case of heterogeneous collections, which I indeed missed.